### PR TITLE
Misc fixes and refactoring in bulletproof module

### PR DESCRIPTION
**Pull request by jonas-lj**

### CODE REVIEW

File: fastcrypto/src/bulletproofs.rs [hunk 1]
```diff
@@ -51,99 +56,22 @@ impl PedersenCommitment {
...
-    fn eq(&self, other: &Self) -> bool {
-        self.point == other.point
+    fn to_byte_array(&self) -> [u8; PEDERSEN_COMMITMENT_LENGTH] {
+        self.point.compress().to_bytes()
     }
 }
 
-impl Eq for PedersenCommitment {}
-
-impl PartialOrd for PedersenCommitment {
```
Comment (jonas-lj) [inline fastcrypto/src/bulletproofs.rs:136]: Are these (`PartialOrd` and `Ord`) really needed?

Comment (joyqvq) [reply fastcrypto/src/bulletproofs.rs:136]: is this for sui later? we dont have to add now

File: fastcrypto/src/bulletproofs.rs [hunk 2]
```diff
@@ -15,29 +15,36 @@
...
+};
+
+use serde::de;
+use serde::Deserialize;
 
 //
 // Pedersen commitments
 //
 
+// TODO: The scalars (value and blinding) are created from the lower 255 bits of a 256 bit scalar, so we should require that the last bit is zero or use the scalars directly.
```
Comment (joyqvq) [inline fastcrypto/src/bulletproofs.rs:41]: 👍

**Review by joyqvq — APPROVED** [review]
lgtm

File: fastcrypto/src/bulletproofs.rs (hunk 1 above)
Comment (jonas-lj) [reply fastcrypto/src/bulletproofs.rs:136]: Lets add them later if they are needed.

**Pull request merged**